doc/adr/0002-pab-indexing-solution-integration.rst

```diff
@@ -0,0 +1,142 @@
+
+Meanwhile, plutus-chain-index (our main indexing solution at the time of this writing) is also listening to incoming blocks from the local node and indexes them into a database. The indexed data can be queried using the REST API interface.
+
+This brings up the main issue: the PAB and plutus-chain-index each listen to the same source of information (a local Cardano node), but each index the information at different speeds. For a dApp developer writing off-chain code using the Contract API, there is no abstraction for handling multiple sources of truth.
+
+Currently, in the best case scenario (fully synced PAB and plutus-chain-index), plutus-chain-index will always trail behind the in-memory storage of the PAB by a few seconds. Therefore, even in this scenario, querying the plutus-chain-index with `unspentTxOutFromRef` in the above contract has a high probability of returning `Nothing`.
+
+Decision
+--------
+
+TBD
+
+Alternatives
+------------
+
+Make sure all components are in sync
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We can make sure that all indexing solutions are syncing at the same speed. For example, if PAB syncs from the local node and arrives at slot 100, then it needs to wait for the chain-index to also arrive at slot 100. Only then can it respond to a Contract request.
+
+To be able to do that, we would need to modify the block listener of PAB and plutus-chain-index to only get new blocks once they are at the same slot.
+
+Pros:
+
+- No modification to the Contract API
+- All the indexed information is consistent with each other
+
+Cons:
+
+- Hard to implement
+- As fast as the slowest indexing solution
+- Can not be feasible with external indexing solutions which we can’t control the indexing speed (e.g. Blockfrost_ or Scrolls_)
+
+Add indexing specific functions in the Contract API
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In this scenario, we would need to split Contract API requests which interact with an external indexing solution to the ones that use the PAB. Currently, we have `awaitTxConfirmed` which uses the indexed information in the PAB to wait for a transaction status to change to `Confirmed`. On top of that, we can have `awaitTxIndexed` or `awaitTxOutIndexed` which will wait for the information to be indexed in the external indexing solution.
+
+Pros:
+
+- Limits design change on the PAB
+- More control given to the user of the Contract API
+
+Cons:
+
+- Adds (undesirable?) complexity to the Contract API
+- We'll need to add a bunch of functions (ex. `awaitMarconiTxConfirmed or `awaitScrollsTxConfirmed`) for each new indexing solution we want to support
```

Comment:
recommend using either "e.g." or "for example" instead of ex. --> 

(e.g., 'awaitMarconiTxConfirmed' or ... ) 
(for example, 'awaitMarconiTxConfirmed' or ... )